src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs

```diff
@@ -282,16 +292,31 @@ public virtual bool PassesFinalFilters(TreeNode parent, object context)
             return true;
         }
 
-        public override string GetNodeSubType(object context)
+        public override string GetNodeSubType(object smoObject, SmoQueryContext smoContext)
         {
             return string.Empty;
         }
 
-        public override string GetNodeStatus(object context)
+        public override string GetNodeStatus(object smoObject, SmoQueryContext smoContext)
         {
             return string.Empty;
         }
 
+        public static bool IsPropertySupported(string propertyName, SmoQueryContext context, NamedSmoObject smoObj, IEnumerable<NodeSmoProperty> supportedProperties)
+        {
+            var property = supportedProperties.FirstOrDefault(x => x.Name == propertyName);
+            if (property != null)
+            {
+                return ServerVersionHelper.IsValidFor(context.ValidFor, property.ValidFor);
+            }
+            else
+            {
+                // Return true if cannot find the proeprty, SMO still tries to get that property but adding the proeprty to supported list can make loading the nodes faster
+                Logger.Write(LogLevel.Verbose, $"Smo property name {propertyName} for Smo type {smoObj.GetType()} is not added as supported properties. This can cause the performance of loading the OE nodes");

```

Comment:
Nice. Any way we could catch errors like this during automatic testing?

Reply:
o I don't have a good way to do this. I'll think about it 

---
In reply to: [154442107](https://github.com/Microsoft/sqltoolsservice/pull/555#discussion_r154442107) [](ancestors = 154442107)